code/modules/mob/living/basic/guardian/guardian_types/gaseous.dm

```diff
@@ -0,0 +1,153 @@
+	/// Rate of temperature stabilization per second.
+	var/temp_stabilization_rate = 0.1
+
+/mob/living/basic/guardian/gaseous/Initialize(mapload, theme)
+	. = ..()
+	RegisterSignal(src, COMSIG_ATOM_PRE_PRESSURE_PUSH, PROC_REF(pre_pressure_moved))
+	gas = new(src)
+	gas.owner_has_control = FALSE // It's nicely integrated with the Guardian UI, no need to have two buttons
+	gas.Grant(src)
+
+/mob/living/basic/guardian/gaseous/Destroy()
+	QDEL_NULL(gas)
+	return ..()
+
+/mob/living/basic/guardian/gaseous/toggle_modes()
+	gas.Trigger()
+
+/mob/living/basic/guardian/gaseous/Life(seconds_per_tick, times_fired)
+	. = ..()
+	if (isnull(summoner))
+		return
+	summoner.extinguish_mob()
+	summoner.set_fire_stacks(0, remove_wet_stacks = FALSE)
+	summoner.adjust_bodytemperature(get_temp_change_amount((summoner.get_body_temp_normal() - summoner.bodytemperature), temp_stabilization_rate * seconds_per_tick))
+
+/mob/living/basic/guardian/gaseous/melee_attack(atom/target, list/modifiers, ignore_cooldown)
+	. = ..()
+	if(!. || !isliving(target))
+		return
+	do_sparks(1, TRUE, target)
+
+/mob/living/basic/guardian/gaseous/recall_effects()
+	. = ..()
+	if(!isnull(summoner))
+		UnregisterSignal(summoner, COMSIG_ATOM_PRE_PRESSURE_PUSH)
+
+/mob/living/basic/guardian/gaseous/manifest_effects()
+	. = ..()
+	if (!isnull(summoner))
+		RegisterSignal(summoner, COMSIG_ATOM_PRE_PRESSURE_PUSH, PROC_REF(pre_pressure_moved))
+
+/// We stand firm in the face of gas
+/mob/living/basic/guardian/gaseous/proc/pre_pressure_moved(datum/source)
+	SIGNAL_HANDLER
+	return COMSIG_ATOM_BLOCKS_PRESSURE
+
+
+/// Expel a range of gases
+/datum/action/cooldown/mob_cooldown/expel_gas
+	name = "Release Gas"
+	desc = "Start or stop expelling a selected gas into the environment."
+	button_icon = 'icons/mob/actions/actions_spells.dmi'
+	button_icon_state = "smoke"
+	cooldown_time = 0 SECONDS // We're here for the interface not the cooldown
+	melee_cooldown_time = 0 SECONDS
+	click_to_activate = FALSE
+	/// Gas being expelled.
+	var/active_gas = null
+	/// Associative list of types of gases to moles we create every life tick.
+	var/static/list/possible_gases = list(
+		/datum/gas/oxygen = 50,
+		/datum/gas/nitrogen = 750, //overpressurizing is hard!.
+		/datum/gas/water_vapor = 1, //you need incredibly little water vapor for the effects to kick in
+		/datum/gas/nitrous_oxide = 15,
+		/datum/gas/carbon_dioxide = 50,
+		/datum/gas/plasma = 3,
+		/datum/gas/bz = 10,
+	)
+	/// Associative list of types of gases to particle effect colour.
+	var/static/list/gas_colors = list(
+		/datum/gas/oxygen = "#63BFDD", //color of frozen oxygen
+		/datum/gas/nitrogen = "#777777", //grey (grey)
+		/datum/gas/water_vapor = "#96ADCF", //water is slightly blue
+		/datum/gas/nitrous_oxide = "#FEFEFE", //white like the sprite
+		/datum/gas/carbon_dioxide = "#222222", //black like coal
+		/datum/gas/plasma = "#B233CC", //color of the plasma sprite
+		/datum/gas/bz = "#FAFF00", //color of the bz metabolites reagent
+	)
```

Comment:
don't we have color constants for these in the code now after ninja added them for pipe gas visuals? they match up to TGUI iirc